components/org.wso2.carbon.identity.application.authenticator.samlsso/src/main/java/org/wso2/carbon/identity/application/authenticator/samlsso/logout/validators/LogoutRequestValidator.java

```diff
@@ -0,0 +1,188 @@
+                String notification = "Invalid Issuer Format in the logout request";
+                if (log.isDebugEnabled()) {
+                    log.debug(notification);
+                }
+                samlMessageContext.setValidStatus(false);
+                String errorResponse = SAMLFedLogoutUtil.buildErrorResponse(samlMessageContext, logoutRequest.getID(),
+                        SSOConstants.StatusCodes.REQUESTOR_ERROR, notification);
+                throw new SAMLIdentityException(notification, errorResponse, logoutRequest.getDestination(),
+                        samlMessageContext.getRelayState());
+            }
+        } else {
+            String notification = "Issuer value cannot be null in the Logout Request";
+            if (log.isDebugEnabled()) {
+                log.debug(notification);
+            }
+            samlMessageContext.setValidStatus(false);
+            String errorResponse = SAMLFedLogoutUtil.buildErrorResponse(samlMessageContext, logoutRequest.getID(),
+                    SSOConstants.StatusCodes.REQUESTOR_ERROR, notification);
+            throw new SAMLIdentityException(notification, errorResponse, logoutRequest.getDestination(),
+                    samlMessageContext.getRelayState());
+        }
+    }
+
+    /**
+     * Validate the subject of the logout request.
+     *
+     * @param logoutRequest {@link LogoutRequest} object to be validated.
+     * @throws SAMLIdentityException If Subject of the Logout Request validation fails.
+     */
+    private void isSubjectValid(LogoutRequest logoutRequest) throws SAMLIdentityException {
+
+        if (logoutRequest.getNameID() == null && logoutRequest.getBaseID() == null
+                && logoutRequest.getEncryptedID() == null) {
+            String notification = "Subject Name should be specified in the Logout Request";
+            if (log.isDebugEnabled()) {
+                log.debug(notification);
+            }
+            samlMessageContext.setValidStatus(false);
+            String errorResponse = SAMLFedLogoutUtil.buildErrorResponse(samlMessageContext, logoutRequest.getID(),
+                    SSOConstants.StatusCodes.REQUESTOR_ERROR, notification);
+            throw new SAMLIdentityException(notification, errorResponse, logoutRequest.getDestination(),
+                    samlMessageContext.getRelayState());
+        }
+    }
+
+    /**
+     * Validate the signature of the logout request.
+     *
+     * @param logoutRequest {@link LogoutRequest} object to be validated.
+     * @throws SAMLIdentityException If Signature of the Logout Request validation fails.
+     */
+    private void isValidLogoutReqSignature(LogoutRequest logoutRequest) throws SAMLIdentityException {
+
+        if (TRUE.equals(samlMessageContext.getFedIdPConfigs().get(IS_LOGOUT_REQ_SIGNED))) {
+            if (!SAMLFedLogoutUtil.isValidSignature(logoutRequest, samlMessageContext)) {
```

Comment:
Without having two if blocks we can improve this logic as follows.
if (TRUE.equals(samlMessageContext.getFedIdPConfigs().get(IS_LOGOUT_REQ_SIGNED)) && 
!SAMLFedLogoutUtil.isValidSignature(logoutRequest, samlMessageContext))